src/tasks/component-installers/cert-manager.ts

```diff
@@ -0,0 +1,182 @@
+
+export class CertManagerTasks {
+  protected kubeHelper: KubeHelper
+
+  constructor(flags: any) {
+    this.kubeHelper = new KubeHelper(flags)
+  }
+
+  /**
+   * Returns list of tasks which perform cert-manager checks and deploy and requests self-signed certificate for Che.
+   */
+  getTasks(flags: any): ReadonlyArray<Listr.ListrTask> {
+    return [
+      {
+        title: 'Check Cert Manager deployment',
+        task: async (ctx: any, task: any) => {
+          // Check only one CRD of cert-manager assuming that it is installed or not.
+          ctx.certManagerInstalled = await this.kubeHelper.namespaceExist(CERT_MANAGER_NAMESPACE_NAME) && await this.kubeHelper.crdExist('certificates.cert-manager.io')
+          if (ctx.certManagerInstalled) {
+            task.title = `${task.title}...already deployed`
+          } else {
+            task.title = `${task.title}...not deployed`
+          }
+        }
+      },
+      {
+        title: 'Deploy cert-manager',
+        enabled: ctx => !ctx.certManagerInstalled,
+        task: async (ctx: any, task: any) => {
+          const yamlPath = path.join(flags.templates, 'cert-manager', 'cert-manager.yml')
+          await this.kubeHelper.applyResource(yamlPath)
+          ctx.certManagerInstalled = true
+
+          task.title = `${task.title}...done`
+        }
+      },
+      {
+        title: 'Wait for cert-manager',
+        enabled: ctx => ctx.certManagerInstalled,
+        task: async (ctx: any, task: any) => {
+          if (!ctx.certManagerInstalled) {
+            throw new Error('Cert Manager should be deployed before.')
+          }
+
+          await this.kubeHelper.waitForPodReady('app.kubernetes.io/name=cert-manager', CERT_MANAGER_NAMESPACE_NAME)
+          await this.kubeHelper.waitForPodReady('app.kubernetes.io/name=webhook', CERT_MANAGER_NAMESPACE_NAME)
+          await this.kubeHelper.waitForPodReady('app.kubernetes.io/name=cainjector', CERT_MANAGER_NAMESPACE_NAME)
+
+          task.title = `${task.title}...ready`
+        }
+      },
+      {
+        title: 'Check Cert Manager CA certificate',
+        task: async (ctx: any, task: any) => {
+          if (!ctx.certManagerInstalled) {
+            throw new Error('Cert manager must be installed before.')
+          }
+          // To be able to use self-signed sertificate it is required to provide CA private key & certificate to cert-manager
+          const caSelfSignedCertSecret = await this.kubeHelper.getSecret(CERT_MANAGER_CA_SECRET_NAME, CERT_MANAGER_NAMESPACE_NAME)
+          if (!caSelfSignedCertSecret) {
+            // First run, generate CA self-signed certificate
+
+            task.title = `${task.title}...generating new one`
+
+            const CA_CERT_GENERATION_SERVICE_ACCOUNT_NAME = 'ca-cert-generator'
+            const CA_CERT_GENERATION_JOB_NAME = 'ca-cert-generation-job'
+            const CA_CERT_GENERATION_JOB_IMAGE = 'quay.io/eclipse/che-cert-manager-ca-cert-generator:latest'
```

Comment:
Yes, thants the plan for next PR. I want to move dockerfile into Che repository and automate the build.

Reply:
Could this be based on UBI instead of Alpine to make it a bit easier to productize? That way you could potentially RUN this build on demand rather than having Yet Another Container To Build, Publish, Maintain, Keep CVE Free, and Productize.

Reply:
I guess we can't build on demand but moving to UBI would still make downstream easier. Or else I can contribute a rhel.Dockerfile variant that we can use in CRW.